src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -286,6 +289,16 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
 
             _recentItemsManager.MakeMostRecentItem(roslynItem.FilterText);
 
+            if (provider is INotifyCommittingItemCompletionProvider notifyProvider)
+            {
+                _ = ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    // Make sure the notification isn't sent on UI thread.
+                    await TaskScheduler.Default;
+                    await notifyProvider.NotifyCommittingItemAsync(document, roslynItem, commitCharacter, cancellationToken).ConfigureAwait(false);
+                });
```

Comment:
should we track exceptions with a reportnonfatalerror since this is fire and forget?  

Also for tests, should this also be covered under an async listener - it also might make the test simpler as you can just wait for the async operation to complete

Reply:
Makes sense, fixed. As of the test, since there's only one, I will just keep it as is.